### PR TITLE
[DRILL-7847] Give all maven modules a consistent name

### PR DESCRIPTION
**Pull request by nielsbasjes**
# [DRILL-7847](https://issues.apache.org/jira/browse/DRILL-7847): Give all maven modules a consistent name

## Description

When building Drill the maven output shows the various modules with names that are inconsistent and hard to read.
Some examples of what is there now:
```
[INFO] Apache Drill Root POM .............................. SUCCESS [  0.100 s]
[INFO] tools/Parent Pom ................................... SUCCESS [  0.003 s]
[INFO] tools/freemarker codegen tooling ................... SUCCESS [  0.002 s]
[INFO] Drill Protocol ..................................... SUCCESS [  0.002 s]
[INFO] Common (Logical Plan, Base expressions) ............ SUCCESS [  0.002 s]
[INFO] Logical Plan, Base expressions ..................... SUCCESS [  0.018 s]
[INFO] exec/Parent Pom .................................... SUCCESS [  0.002 s]
...
[INFO] exec/JDBC Driver using dependencies ................ SUCCESS [  0.001 s]
[INFO] JDBC JAR with all dependencies ..................... SUCCESS [  0.037 s]
[INFO] Drill-on-YARN ...................................... SUCCESS [  0.002 s]
[INFO] metastore/Drill RDBMS Metastore .................... SUCCESS [  0.004 s]
[INFO] contrib/kudu-storage-plugin ........................ SUCCESS [  0.030 s]
```

I propose to have a consistent naming of all modules that is easier for the developer to read and understand.

## Documentation
None.

## Testing
If a developer runs the build the naming of the modules is easier to read and understand.

The initial version here outputs this on a `mvn clean`

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Drill : 1.19.0-SNAPSHOT:
[INFO] 
[INFO] Drill : ............................................ SUCCESS [  0.111 s]
[INFO] Drill : Tools : .................................... SUCCESS [  0.002 s]
[INFO] Drill : Tools : Freemarker codegen ................. SUCCESS [  0.002 s]
[INFO] Drill : Protocol ................................... SUCCESS [  0.003 s]
[INFO] Drill : Common ..................................... SUCCESS [  0.003 s]
[INFO] Drill : Logical Plan, Base expressions ............. SUCCESS [  0.006 s]
[INFO] Drill : Exec : ..................................... SUCCESS [  0.003 s]
[INFO] Drill : Exec : Memory : ............................ SUCCESS [  0.002 s]
[INFO] Drill : Exec : Memory : Base ....................... SUCCESS [  0.003 s]
[INFO] Drill : Exec : RPC ................................. SUCCESS [  0.002 s]
[INFO] Drill : Exec : Vectors ............................. SUCCESS [  0.002 s]
[INFO] Drill : Contrib : .................................. SUCCESS [  0.003 s]
[INFO] Drill : Contrib : Data : ........................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Data : TPCH Sample ............... SUCCESS [  0.049 s]
[INFO] Drill : Metastore : ................................ SUCCESS [  0.002 s]
[INFO] Drill : Metastore : API ............................ SUCCESS [  0.002 s]
[INFO] Drill : Metastore : Iceberg ........................ SUCCESS [  0.001 s]
[INFO] Drill : Exec : Java Execution Engine ............... SUCCESS [  0.002 s]
[INFO] Drill : Exec : JDBC Driver using dependencies ...... SUCCESS [  0.002 s]
[INFO] Drill : Exec : JDBC JAR with all dependencies ...... SUCCESS [  0.036 s]
[INFO] Drill : Hadoop YARN ................................ SUCCESS [  0.002 s]
[INFO] Drill : Metastore : RDBMS Metastore ................ SUCCESS [  0.004 s]
[INFO] Drill : Contrib : Storage : Kudu ................... SUCCESS [  0.030 s]
[INFO] Drill : Contrib : Format : XML ..................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : HTTP ................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : OpenTSDB ............... SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Storage : Mongo .................. SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : HBase .................. SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : JDBC ................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : Hive : ................. SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Storage : Hive : Exec shaded ..... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : Hive : Core ............ SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Storage : Kafka .................. SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Storage : ElasticSearch .......... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : UDFs ............................. SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : Syslog .................. SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Format : Httpd/Nginx access log .. SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : HDF5 .................... SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Format : SPSS .................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : LTSV .................... SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Format : Image ................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : Esri .................... SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : Excel ................... SUCCESS [  0.001 s]
[INFO] Drill : Contrib : Storage : Druid .................. SUCCESS [  0.002 s]
[INFO] Drill : Packaging and Distribution Assembly ........ SUCCESS [  0.002 s]
[INFO] Drill : Contrib : Format : MaprDB .................. SUCCESS [  0.002 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

```

### CODE REVIEW

**Comment by luocooong** [comment]
@nielsbasjes 
Hi. Is it possible to add a title (and naming method) about the consistent name of module in `/master/docs/dev/DevDocs.md`? Maybe it's friendly for new contributor. thanks

**Comment by nielsbasjes** [comment]
Sure, I'll put up a proposal for you to evaluate.

File: contrib/format-httpd/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <version>1.19.0-SNAPSHOT</version>
   </parent>
   <artifactId>drill-format-httpd</artifactId>
-  <name>contrib/httpd-format-plugin</name>
+  <name>Drill : Contrib : Format : Httpd/Nginx access log</name>
```
Comment (luocooong) [inline contrib/format-httpd/pom.xml:29]: I recommend that `Nginx access log` to `Nginx Access Log`

File: contrib/storage-mongo/pom.xml [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
   <artifactId>drill-mongo-storage</artifactId>
 
-  <name>contrib/mongo-storage-plugin</name>
+  <name>Drill : Contrib : Storage : Mongo</name>
```
Comment (luocooong) [inline contrib/storage-mongo/pom.xml:31]: I recommend that `Mongo` to `MongoDB`

File: logical/pom.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
   <artifactId>drill-logical</artifactId>
   <packaging>jar</packaging>
-  <name>Logical Plan, Base expressions</name>
+  <name>Drill : Logical Plan, Base expressions</name>
```
Comment (luocooong) [inline logical/pom.xml:32]: I recommend that `Drill : Logical Plan`

File: metastore/rdbms-metastore/pom.xml [hunk 4]
```diff
@@ -28,7 +28,7 @@
   </parent>
 
   <artifactId>drill-rdbms-metastore</artifactId>
-  <name>metastore/Drill RDBMS Metastore</name>
+  <name>Drill : Metastore : RDBMS Metastore</name>
```
Comment (luocooong) [inline metastore/rdbms-metastore/pom.xml:31]: I recommend that `RDBMS Metastore` to `RDBMS`

File: drill-yarn/pom.xml [hunk 5]
```diff
@@ -28,7 +28,7 @@
 
   <artifactId>drill-yarn</artifactId>
   <packaging>jar</packaging>
-  <name>Drill-on-YARN</name>
+  <name>Drill : Hadoop YARN</name>
```
Comment (luocooong) [inline drill-yarn/pom.xml:31]: I recommend that `Hadoop YARN` to `On-YARN`

File: contrib/storage-hive/hive-exec-shade/pom.xml [hunk 6]
```diff
@@ -29,7 +29,7 @@
 
   <artifactId>drill-hive-exec-shaded</artifactId>
   <packaging>jar</packaging>
-  <name>contrib/hive-storage-plugin/hive-exec-shaded</name>
+  <name>Drill : Contrib : Storage : Hive : Exec shaded</name>
```
Comment (luocooong) [inline contrib/storage-hive/hive-exec-shade/pom.xml:32]: I recommend that `Exec shaded` to `Exec Shaded`

**Review by luocooong — COMMENTED** [review]
@nielsbasjes Thanks for the contribution. I left a few review comments.

**Comment by luocooong** [comment]
+1

**Comment by nielsbasjes** [comment]
@luocooong I implemented your suggestions.

**Review by luocooong — APPROVED** [review]
@cgivre If changing the name will not affect other modules, I think this PR is very nice.

**Comment by luocooong** [comment]
@nielsbasjes Please squash commits and I think we're ready to merge.

**Comment by nielsbasjes** [comment]
@luocooong Did a rebase on the latest master and squashed the commits.

**Pull request merged**